deploy/examples/realm/realm_with_ldap.yaml

```diff
@@ -0,0 +1,80 @@
+piVersion: keycloak.org/v1alpha1
+kind: KeycloakRealm
+metadata:
+  labels:
+    app: sso
+  name: ldap-realm
+spec:
+  instanceSelector:
+    matchLabels:
+      app: sso
+  realm:
+    displayName: LDAPRealm
+    enabled: true
+    id: ldap-realm
+    realm: ldap-realm
+    userFederationProviders:
+      - displayName: "ldap"
+        providerName: "ldap"
+        config:
+          vendor: "ad"
+          connectionUrl: "ldap://localhost"
+          bindDn: "USERNAME"
+          # This reads the REALM-SECRET from /opt/jboss/keycloak/secrets
+          # In this example, the secret key is called ldap-realm_bindCredential
+          # See deploy/examples/keycloak/keycloak-with-vault.yaml for an
+          # example of how to setup the vault secret directory
+          bindCredential: "${vault.bindCredential}"
+          usersDn: DC=example,DC=com"
+          usernameLDAPAttribute: "mail"
+          uuidLDAPAttribute: "objectGUID"
+          searchScope: "2" # sub
+          useTruststoreSpi: "ldapsOnly"
+          trustEmail: "true"
+          userObjectClasses: "person, organizationalPerson, user"
+          rdnLDAPAttribute: "cn"
+          editMode: "READ_ONLY"
+          # debug: "false"
+    userFederationMappers:
+      - name: username
+        federationProviderDisplayName: ldap
+        federationMapperType: user-attribute-ldap-mapper
+        config:
+          always.read.value.from.ldap: 'true'
+          is.binary.attribute: 'false'
+          is.mandatory.in.ldap: 'true'
+          ldap.attribute: mail
+          read.only: 'true'
+          user.model.attribute: username
+      - name: MSAD account controls
+        federationProviderDisplayName: ldap
+        federationMapperType:  msad-user-account-control-mapper
+        config:
+          ldap.password.policy.hints.enabled: 'false'
+      - name: last name
+        federationProviderDisplayName: ldap
+        federationMapperType: user-attribute-ldap-mapper
+        config:
+          always.read.value.from.ldap: 'true'
+          is.binary.attribute: 'false'
+          is.mandatory.in.ldap: 'true'
+          ldap.attribute: sn
+          read.only: 'true'
+          user.model.attribute: lastName
+      - name: email
+        federationProviderDisplayName: ldap
+        federationMapperType: user-attribute-ldap-mapper
+        config:
+          always.read.value.from.ldap: 'true'
+          is.binary.attribute: 'false'
+          is.mandatory.in.ldap: 'true'
+          ldap.attribute: mail
+          read.only: 'true'
+          user.model.attribute: email
+      - name: full name
+        federationProviderDisplayName: ldap
+        federationMapperType: full-name-ldap-mapper
+        config:
+          ldap.full.name.attribute: cn
+          read.only: 'true'
+          write.only: 'false'
```

Comment:
So I'm understanding this right, and maybe would be good to add to the example, would adding this mapper to pull in ldap groups work here?
```
- name: group
  federationProviderDisplayName: ldap
  federationMapperType: group-ldap-mapper
  config:
    membership.attribute.type: DN
    group.name.ldap.attribute: cn
    membership.user.ldap.attribute: sAMAccountName
    preserve.group.inheritance: 'true'
    groups.dn: DC=example,DC=com
    mode: READ_ONLY
    user.roles.retrieve.strategy: GET_GROUPS_FROM_USER_MEMBEROF_ATTRIBUTE
    groups.ldap.filter: "(cn=EXAMPLE*)" #Only pull in groups that start with prefix EXAMPLE
    membership.ldap.attribute: "true"
    memberof.ldap.attribute: memberOf
    group.object.classes: group
    drop.non.existing.groups.during.sync: "true"
```

Reply:
That's right. I'll add yours as an example. I'm using this one for talking to my openldap server:

```
            - name: "groups"
              federationMapperType: group-ldap-mapper
              config:
                drop.non.existing.groups.during.sync: 'false'
                group.name.ldap.attribute: cn
                group.object.classes: posixGroup
                groups.dn: "ou=Groups,dc=example,dc=com"
                ignore.missing.groups: 'false'
                memberof.ldap.attribute: "memberOf"
                membership.attribute.type: "UID"
                membership.ldap.attribute: "memberUid"
                membership.user.ldap.attribute: "uid"
                mode: "READ_ONLY"
                preserve.group.inheritance: 'false'
                user.roles.retrieve.strategy: "LOAD_GROUPS_BY_MEMBER_ATTRIBUTE"
```